src/formats/internal/reader.c

```diff
@@ -213,6 +247,16 @@ int sd_read_line(const sd_reader *reader, char *buffer, int maxlen) {
     if(fgets(buffer, maxlen, reader->handle) == NULL) {
         return 1;
     }
+#ifdef BIG_ENDIAN_BUILD
+    int len = maxlen;
+    for(int i = 0; i < len / 4; i += 1) {
+        ((int *)buffer)[i] = __builtin_bswap32(((int *)buffer)[i]);
```

Comment:
`char *buffer` is insufficiently aligned to be a `int*`
same comment applies in sd_read_str.

please memcpy to a uint32_t local variable (and uint16_t local var for the final-2-bytes bytes-swap)